### PR TITLE
Fixed null issues with instance labels from template

### PR DESCRIPTION
**Pull request by rachely3n**
When testing https://github.com/jenkinsci/google-compute-engine-plugin/pull/12/ via the UI with an instance template I created, got nullPointerException when no label is provided by template.

Made a fix so that label is optional.

Testing
- tested with templates on both linux and windows with no labels in UI
- ran mvn verify, passed unit and integration tests

### CODE REVIEW

File: src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java [hunk 1]
```diff
@@ -339,12 +339,11 @@ public Instance instance() throws IOException {
         if (StringUtils.isNotEmpty(template)) {
             // TODO: JENKINS-55285
             InstanceTemplate instanceTemplate = cloud.client.getTemplate(nameFromSelfLink(cloud.projectId), nameFromSelfLink(template));
-            Map<String, String> templateLabels = instanceTemplate.getProperties().getLabels();
-            Map<String, String> mergedLabels = new HashMap<>(templateLabels);
-            mergedLabels.putAll(googleLabels);
-            instance.setLabels(mergedLabels);
+            if (instanceTemplate.getProperties().getLabels() != null) {
+                Map<String, String> templateLabels = instanceTemplate.getProperties().getLabels();
+                appendLabels(templateLabels);
```
Comment (ingwarsw) [inline src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java]: You cannot do that like that cause your changing Instance configuration for good..
And you will add labels with every run..

Comment (rachely3n) [reply src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java]: Great catch, I've made a change. Feel free to comment.

**Comment by rachely3n** [comment]
@ingwarsw, could you clarify what you mean by adding labels to all generated instances? Did this PR change any logic?

**Comment by ingwarsw** [comment]
@rachely3n No no no.. it did not change anything.. I just never had that issue cause our templates have to contain a lot of labels :)

**Comment by craigatgoogle** [comment]
It's usually good practice to include a test case testing the use case which caused the bug to manifest. This ensures the fix and protects against future regressions. Suggest adding a case to the IT for this.

**Pull request merged**